### PR TITLE
[NOTICKET] allow zip passwords for sql files

### PR DESCRIPTION
**Pull request by jochenmanz**

### CODE REVIEW

File: devenv.nix [hunk 1]
```diff
@@ -141,35 +143,41 @@ let
...
 
     if [[ "$1" == *.sql ]]; then
-        curl --create-dirs "$1" --output "$TARGETFOLDER/dump.sql"
-    elif [[ "$1" == *.sql.gz ]]; then
-        curl --create-dirs "$1" --output "$TARGETFOLDER/latest.sql.gz"
-        gunzip -c "$TARGETFOLDER/latest.sql.gz" > "$TARGETFOLDER/dump.sql"
-    elif [[ "$1" == *.sql.zip ]]; then
-        curl --create-dirs "$1" --output "$TARGETFOLDER/latest.sql.zip"
-        unzip -j -o "$TARGETFOLDER/latest.sql.zip" '*.sql' -d "$TARGETFOLDER"
+      ${pkgs.curl}/bin/curl --create-dirs "$1" --output "$TARGETFOLDER/dump.sql"
```
Comment (kleinmann) [inline devenv.nix:149]: If we always used `latest.sql` as target file, the logic for SQL_FILE could be simplified.

File: devenv.nix [hunk 2]
```diff
@@ -141,35 +143,41 @@ let
...
+    elif [[ "$1" == *.gz ]]; then
+      ${pkgs.curl}/bin/curl --create-dirs "$1" --output "$TARGETFOLDER/latest.sql.gz"
+      ${pkgs.gzip}/bin/gunzip -c "$TARGETFOLDER/latest.sql.gz" > "$TARGETFOLDER/dump.sql"
+    elif [[ "$1" == *.zip ]]; then
+      ${pkgs.curl}/bin/curl --create-dirs "$1" --output "$TARGETFOLDER/latest.sql.zip"
+
+      if [ x"$IMPORT_DATABASE_PASSWORD" == "x" ]; then
+        UNZIP_COMMAND="${pkgs.unzip}/bin/unzip"
+      else
+        UNZIP_COMMAND="${pkgs.unzip}/bin/unzip -P $IMPORT_DATABASE_PASSWORD"
```
Comment (kleinmann) [inline devenv.nix:159]: I'm not sure if we should add this functionality. It is prone to user error (committing passwords) and brittle (special characters breaking shell expansion).
Is this a must-have for you?

File: devenv.nix [hunk 3]
```diff
@@ -141,35 +143,41 @@ let
...
+      exit
     fi
 
-    LANG=C LC_CTYPE=C LC_ALL=C sed -e 's/DEFINER[ ]*=[ ]*[^*]*\*/\*/' "$SQL_FILE"
-    LANG=C LC_CTYPE=C LC_ALL=C sed 's/NO_AUTO_CREATE_USER//' "$SQL_FILE"
+    LANG=C LC_CTYPE=C LC_ALL=C ${pkgs.gnused}/bin/sed -e 's/DEFINER[ ]*=[ ]*[^*]*\*/\*/' "$SQL_FILE" > $SQL_FILE
+    LANG=C LC_CTYPE=C LC_ALL=C ${pkgs.gnused}/bin/sed 's/NO_AUTO_CREATE_USER//' "$SQL_FILE" > $SQL_FILE
 
-    MYSQL_PWD="" ${config.services.mysql.package}/bin/mysql shopware -f < "$SQL_FILE"
+    MYSQL_PWD="" ${config.services.mysql.package}/bin/mysql -u root shopware -f < "$SQL_FILE"
```
Comment (kleinmann) [inline devenv.nix:180]: Our devenv configuration sets the username for mysql to `shopware`. Why would we need `root` here?

File: devenv.nix [hunk 4]
```diff
@@ -253,14 +259,15 @@ in {
...
-      description = "Sets the matcher paths to be \"ignored\" by caddy";
+      description = ''Sets the matcher paths to be "ignored" by caddy'';
     };
   };
 
   config = lib.mkIf cfg.enable {
     packages = [
       pkgs.jq
       pkgs.gnupatch
+      pkgs.gnused
```
Comment (kleinmann) [inline devenv.nix:270]: Is `curl` already included in devenv or should we add it as well?

**Review by kleinmann — CHANGES_REQUESTED** [review]

File: devenv.nix [hunk 5]
```diff
@@ -141,35 +143,41 @@ let
...
-        echo "No SQL file found"
-        exit
+      echo "No SQL file found"
+      exit
     fi
 
-    LANG=C LC_CTYPE=C LC_ALL=C sed -e 's/DEFINER[ ]*=[ ]*[^*]*\*/\*/' "$SQL_FILE"
-    LANG=C LC_CTYPE=C LC_ALL=C sed 's/NO_AUTO_CREATE_USER//' "$SQL_FILE"
+    LANG=C LC_CTYPE=C LC_ALL=C ${pkgs.gnused}/bin/sed -e 's/DEFINER[ ]*=[ ]*[^*]*\*/\*/' "$SQL_FILE" > $SQL_FILE
+    LANG=C LC_CTYPE=C LC_ALL=C ${pkgs.gnused}/bin/sed 's/NO_AUTO_CREATE_USER//' "$SQL_FILE" > $SQL_FILE
```
Comment (Wysselbie) [inline devenv.nix:178]: Same as above.

**Review by Wysselbie — CHANGES_REQUESTED** [review]

File: devenv.nix (hunk 4 above)
Comment (jochenmanz) [reply devenv.nix:270]: We are using the gnused packages directly, ive removed that one from the package list now.

File: devenv.nix (hunk 1 above)
Comment (jochenmanz) [reply devenv.nix:149]: The SQL File inside of the archives could be named differently. But ive renamed dump.sql to latest.sql for a small optimization.

File: devenv.nix (hunk 2 above)
Comment (jochenmanz) [reply devenv.nix:159]: It would streamline some processes, ive removed it for now for a more integrated solution.

File: devenv.nix (hunk 3 above)
Comment (jochenmanz) [reply devenv.nix]: the package does not use the provided configurations. Its fixed now.

File: devenv.nix [hunk 6]
```diff
@@ -141,35 +143,41 @@ let
...
     if [[ "$SQL_FILE" == "" ]]; then
-        echo "No SQL file found"
-        exit
+      echo "No SQL file found"
+      exit
     fi
 
-    LANG=C LC_CTYPE=C LC_ALL=C sed -e 's/DEFINER[ ]*=[ ]*[^*]*\*/\*/' "$SQL_FILE"
-    LANG=C LC_CTYPE=C LC_ALL=C sed 's/NO_AUTO_CREATE_USER//' "$SQL_FILE"
+    LANG=C LC_CTYPE=C LC_ALL=C ${pkgs.gnused}/bin/sed -e 's/DEFINER[ ]*=[ ]*[^*]*\*/\*/' "$SQL_FILE" > $SQL_FILE
```
[the remark this answers is not in the document]
Comment (jochenmanz) [reply devenv.nix]: fixed :)

File: README.md [hunk 7]
```diff
@@ -56,6 +56,26 @@ kellerkinder.additionalPhpExtensions = [
 ];
 ```
 
+### kellerkinder.additionalCaddyServerAlias
+Define additional server aliases that are added to caddy.
+
+*_Example_*
+```
+kellerkinder.additionalCaddyServerAlias = [
```
Comment (kleinmann) [inline README.md:64]: We can omit "Caddy" in the options, since we don't offer any other webservers in our environment 🙂

File: devenv.nix [hunk 8]
```diff
@@ -126,54 +130,49 @@ let
...
   importDbHelper = pkgs.writeScript "importDbHelper" ''
-    PATH="${lib.makeBinPath [ pkgs.coreutils ]}:$PATH"
-
     if [[ "$1" == "" ]]; then
         echo "Please set devenv configuration for kellerkinder.importDatabaseDumps"
         exit
     fi
 
-    if ! ${config.services.mysql.package}/bin/mysqladmin ping > /dev/null 2>&1; then
+    if ! $DEVENV_PROFILE/bin/mysqladmin ping > /dev/null 2>&1; then
```
Comment (kleinmann) [inline devenv.nix:138]:
```suggestion
    if ! "${config.env.DEVENV_PROFILE}/bin/mysqladmin" ping > /dev/null 2>&1; then
```
Shouldn't it be more like this?

**Review by kleinmann — DISMISSED** [review]

File: README.md (hunk 7 above)
Comment (jochenmanz) [reply README.md]: dont :)

**Pull request merged**